src/murmur/main.cpp

```diff
@@ -285,153 +387,87 @@ int main(int argc, char **argv) {
-			i++;
-			inifile = args.at(i);
-		} else if ((arg == "-wipessl")) {
-			wipeSsl = true;
-		} else if ((arg == "-wipelogs")) {
-			wipeLogs = true;
-		} else if ((arg == "-fg")) {
-			detach = false;
-		} else if ((arg == "-v")) {
-			bVerbose = true;
-		} else if ((arg == "-version") || (arg == "--version")) {
-			// Print version and exit (print to regular std::cout to avoid adding any useless meta-information from
-			// using e.g. qWarning
-			std::cout << "Mumble server version " << Version::getRelease().toStdString() << std::endl;
-			return 0;
-		} else if (args.at(i) == QLatin1String("-license") || args.at(i) == QLatin1String("--license")) {
+	if (cli_options.quit)
+		return 0;
+
+	if (cli_options.print_license) {
 #ifdef Q_OS_WIN
-			AboutDialog ad(nullptr, AboutDialogOptionsShowLicense);
-			ad.exec();
-			return 0;
+		AboutDialog ad(nullptr, AboutDialogOptionsShowLicense);
+		ad.exec();
+		return 0;
 #else
-			qInfo("%s\n", qPrintable(License::license()));
-			return 0;
+		qInfo("%s\n", qPrintable(License::license()));
+		return 0;
+		bool readPw = false;
 #endif
-		} else if (args.at(i) == QLatin1String("-authors") || args.at(i) == QLatin1String("--authors")) {
+	} else if (cli_options.print_authors) {
 #ifdef Q_OS_WIN
-			AboutDialog ad(nullptr, AboutDialogOptionsShowAuthors);
-			ad.exec();
-			return 0;
+		AboutDialog ad(nullptr, AboutDialogOptionsShowAuthors);
+		ad.exec();
+		return 0;
 #else
-			qInfo("%s\n",
-				  "For a list of authors, please see https://github.com/mumble-voip/mumble/graphs/contributors");
-			return 0;
+		qInfo("%s\n", "For a list of authors, please see https://github.com/mumble-voip/mumble/graphs/contributors");
+		return 0;
 #endif
-		} else if (args.at(i) == QLatin1String("-third-party-licenses")
-				   || args.at(i) == QLatin1String("--third-party-licenses")) {
+	} else if (cli_options.print_3rd_party_licenses) {
 #ifdef Q_OS_WIN
-			AboutDialog ad(nullptr, AboutDialogOptionsShowThirdPartyLicenses);
-			ad.exec();
-			return 0;
+		AboutDialog ad(nullptr, AboutDialogOptionsShowThirdPartyLicenses);
+		ad.exec();
+		return 0;
 #else
-			qInfo("%s", qPrintable(License::printableThirdPartyLicenseInfo()));
-			return 0;
-#endif
-		} else if ((arg == "-h") || (arg == "-help") || (arg == "--help")) {
-			detach = false;
-			qInfo("Usage: %s [-ini <inifile>] [-supw <password>]\n"
-				  "  --version              Print version information and exit\n"
-				  "  -ini <inifile>         Specify ini file to use.\n"
-				  "  -supw <pw> [srv]       Set password for 'SuperUser' account on server srv.\n"
-#ifdef Q_OS_UNIX
-				  "  -readsupw [srv]        Reads password for server srv from standard input.\n"
+		qInfo("%s", qPrintable(License::printableThirdPartyLicenseInfo()));
+		return 0;
 #endif
-				  "  -disablesu [srv]       Disable password for 'SuperUser' account on server srv.\n"
+	}
+
+	detach = boost::logic::indeterminate(cli_options.detach) ? detach : static_cast< bool >(cli_options.detach);
```

Comment:
Why don't we handle the default value by initializing `cli_options.detach` accordingly?

Reply:
Was not quite finished with refactoring these things.  Looks like I missed a few places hiding behind macros :D